include/dsn/c/app_model.h

```diff
@@ -292,51 +292,54 @@ enum dsn_chkpt_apply_mode
     DSN_CHKPT_LEARN
 };
 
-typedef dsn_error_t (*dsn_app_checkpoint)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef void(*dsn_app_on_batched_rpc_requests)(
+    void*,           ///< context from dsn_app_create
+    int64_t,         ///< ballot
+    int64_t,         ///< decree
+    dsn_message_t*,  ///< request array ptr
+    int              ///< request count
+    );
+
+typedef int(*dsn_app_get_internal_error)(
+    void*     ///< context from dsn_app_create
     );
 
-typedef dsn_error_t(*dsn_app_checkpoint_async)(
-    void*,  ///< context from dsn_app_create
-    int64_t ///< checkpoint version
+typedef dsn_error_t(*dsn_app_sync_checkpoint)(
```

Comment:
ok

On Tue, Jun 21, 2016 at 11:13 AM, Zhenyu Guo <notifications@github.com>
wrote:

> In include/dsn/c/app_model.h
> <https://github.com/imzhenyu/rDSN/pull/403#discussion_r67802265>:
>
> >      );
> >
> > -typedef dsn_error_t(*dsn_app_checkpoint_async)(
> > -    void*,  ///< context from dsn_app_create
> > -    int64_t ///< checkpoint version
> > +typedef dsn_error_t(*dsn_app_sync_checkpoint)(
>
> Well, I think easier-to-read (in this case, know what are the callbacks
> grouped into) is more important than "ugly". Anyway, this is a taste
> problem, and you may prefer what you like for now while we can do it
> consistently later after the merge.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/403/files/89a41d5e513a99014291d1ad76187ae04459b1ab#r67802265>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPRKuzyj677qvKvpz90K6nIMi2LIMks5qN1bUgaJpZM4I2J8D>
> .
>